### PR TITLE
Add the -set_exit_status flag

### PR DESCRIPTION
**Pull request by polyfloyd**
To make this change possible, I moved message handling to the main function, this should also make future modifications easier.

### CODE REVIEW

File: prealloc.go [hunk 1]
```diff
@@ -304,16 +318,8 @@ func (v *returnsVisitor) Visit(node ast.Node) ast.Visitor {
 		return v
 
 	}
-	// if simple is true, then we actually have to check if we had returns inside of our loop. Otherwise, we can just report all messages.
-	if v.simple == true && !v.returnsInsideOfLoop {
```
Comment (alexkohler) [inline prealloc.go]: It looks like we've lost our processing of `v.simple`? On a side note, thanks for taking on this issue! 👍

File: prealloc.go [hunk 2]
```diff
@@ -304,16 +318,8 @@ func (v *returnsVisitor) Visit(node ast.Node) ast.Visitor {
...
-		for _, preallocMsg := range v.preallocMsgs {
-			log.Printf(preallocMsg)
-		}
-	} else if v.simple == false {
-		for _, preallocMsg := range v.preallocMsgs {
-			log.Printf(preallocMsg)
-		}
-	}
+	// if simple is true, then we actually have to check if we had returns
```
Comment (alexkohler) [inline prealloc.go]: Since we're doing processing on 116 now, I think this comment can go?

File: prealloc.go [hunk 3]
```diff
@@ -379,3 +389,15 @@ func (v *returnsVisitor) handleLoops(blockStmt *ast.BlockStmt) {
 	}
 
 }
+
+// Hint stores the infomration about an occurance of a slice that could be
```
Comment (alexkohler) [inline prealloc.go]: infomration -> information, occurance -> occurrence

File: prealloc.go (hunk 2 above)
Comment (polyfloyd) [reply prealloc.go]: Ah, yes it can.

Comment (polyfloyd) [reply prealloc.go]: And so can that `default`

**Comment by alexkohler** [comment]
So I've noticed something a little interesting, I was running your branch on the Go source and noticed a difference between what happens with a build of master:

Master:
```Bash
$ ./prealloc -simple=false ~/go1.9.2/src/... &> masterOutput && wc masterOutput
     104     521    9047 masterOutput
```

This branch:
```Bash
$ ./prealloc -simple=false ~/go1.9.2/src/... &> exitStatusBranchOutput && wc exitStatusBranchOutput
  102071  408389 8638849 exitStatusBranchOutput
```

I'm gonna have to dig a little deeper into what's going on here - my suspicion was duplicates but even after filtering out duplicates, there's still a large difference.

```Bash
$ cat exitStatusBranchOutput | uniq | wc
   94364  377561 7904154
```

**Comment by polyfloyd** [comment]
Try `sort -u`, `uniq` only deduplicates lines that are adjacent to each other.

**Comment by alexkohler** [comment]
```
$ cat exitStatusBranchOutput | sort -u | wc
      98     497    8495
 $ cat masterOutput | sort -u | wc
      98     497    8495
```

Good call! :) I think we're still adding duplicates (on the order of 1000x more lines) as a result of the refactor - once we get that ironed out we'll be good to go!
```
$ cat exitStatusBranchOutput  | wc
  102071  408389 8638849

$ cat masterOutput | wc
     104     521    9047
```

**Comment by polyfloyd** [comment]
That should do it.

**Comment by alexkohler** [comment]
Aha, that took care of it! Thank you again for picking this up 👍

**Pull request merged**